tests/test_project.py

```diff
@@ -74,6 +74,12 @@ def test_root_directory(self):
     def test_workspace_directory(self):
         self.assertEqual(self._tmp_wd, self.project.workspace())
 
+    def test_config_modification(self):
```

Comment:
```suggestion
    def test_config_modification(self):
        # In-memory modification of the project configuration is 
        # deprecated as of 1.3, and will be removed in version 2.0.
        # This unit test should reflect that change beginning 2.0,
        # and check that the project configuration is immutable.
```

Let's make it easier for future devs. :)